js/create-baloons.js

```diff
@@ -81,22 +65,32 @@ const smallIcon = L.icon({
   iconAnchor: [26, 52],
 });
 
+const removeBaloons = () => {
+  document.querySelectorAll('[src="./img/pin.svg"]').forEach((marker) => marker.remove());
+};
 
 // Размещаем маркеры карточек
 const createBaloons = (cards) => {
-  cards.forEach((element) => {
-    const marker = L.marker(
-      {
-        lat: element.location.lat,
-        lng: element.location.lng,
-      },
-      {
-        icon: smallIcon,
-      },
-    );
-    marker.addTo(map)
-      .bindPopup(createPopup(element));
-  });
+  cards
+    .filter(filterTypes)
+    .filter(filterRooms)
+    .filter(filterGuests)
+    .filter(filterPrice)
+    .filter(filterfeatures)
+    .slice(0, AMOUNT_USERS)
```

Comment:
не очень хорошая идея вызывать много раз фильтр, всё можно сделть и за один проход по массиву, просто на каждой итерации нужно проверять на  соответствие сразу всем условиям, это в итоге должно выглядеть как то так 
```suggestion
  cards
    .filter((offer) => filterTypes(offer) && filterRooms(offer) &&  filterGuests(offer ) && filterPrice(offer) && filterFeatures(offer))
    .slice(0, AMOUNT_USERS)
```